plugins/inputs/logfile/tailersrc.go

```diff
@@ -76,6 +76,8 @@ type tailerSrc struct {
 	startTailerOnce sync.Once
 	cleanUpFns      []func()
 }
+// Verify tailerSrc implements LogSrc
+var _ logs.LogSrc = (*tailerSrc)(nil)
```

Comment:
For this, what happens if we don't implement LogSrc? Does this fail the build? I think that's fair. You could make a unit test that creates a tailerSrc and then cast it to LogSrc, and verify it casted ok:
```go
_, ok := someTailer.(logs.LogSrc)
assert.True(t, ok)
```
or something like that. Putting it in here looks a little weird to me, but not a blocker.

Reply:
If `tailerSrc` did not fully implement `logs.LogSrc` then this would be a compile time error.
I think compile time error is preferable to a seperate test.

To be honest, I was iffy about adding this because it seems counterproductive to Golang not requiring a class to explicitly list out the interfaces it implements.

In this case it actually helped me realize that the interface was not being met, so it definitely provided value.

Reply:
Makes sense to me. I'm fine keeping it like this and thanks for the link for context.